pages/runtime/data-api.md

```diff
@@ -0,0 +1,672 @@
+# Data API Service
+
+## Introduction
+Resin's Data API Service constitutes a central node in resin architecture, since it provides interconnection between resin services and communication with the database. At the same time, it exposes an HTTP interface, based on Open Data ([OData](http://www.odata.org)) format, that provides a means to the user to interact with resin resources and fetch or update information regarding their applications, devices and environment variables.
+
+This document aims to describe the endpoints that are provided by the Data API, as well as showing examples of their expected usage.
+
+The requests are expected to follow the OData format, and the responses are provided in JSON format. The authorization token that is required in each request can be found under 'Preferences' in the user's dashboard.
+
+### Base URL
+The base URL for accessing the Data API is:
+
+`https://api.resin.io/v1/`
+
+Building the full URLs for accessing resources requires to append the corresponding endpoints to the base URL.
+
+## Interacting with resources
+
+### Resource: Application
+#### Get all applications
+* Summary: Get all applications along with the corresponding information (application name, id, device type, git repository, latest commit)
+* Endpoint: `/application`
+* Method: `GET`
+* cURL Example:
+
+```
+curl "https://api.resin.io/v1/application"
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <auth token>"
+```
+
+#### Get application by ID
+* Summary: Get a single application by ID
+* Endpoint: `/application(<ID>)`
+* Method: `GET`
+
+Param | Type | Description
+-------|------|------------:
+ID | `INTEGER` | application ID
+
+* cURL Example:
+
+```
+curl "https://api.resin.io/v1/application(<ID>)"
+-H "Content-Type: application/json"
+-H "Authorization: Bearer <auth_token>"
+```
+
+#### Get application by name
+* Summary: Get application with the specified name
+* Endpoint: `/application?$filter=app_name eq '<name>'`
+* Method: `GET`
+
+Param | Type | Description
+------|------|------------:
+name | `STRING` | application name
+
+* cURL Example:
+
+```
+curl "https://api.resin.io/v1/application?\$filter=app_name%20eq%20'<name>'" 
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <auth_token>"
+```
+
+#### Create application
+* Summary: Create an application by specifying the name and the device type.
+* Endpoint: `/application`
+* Method: `POST` 
+
+Param | Type | Description
+------|------|------------:
+app_name | `STRING` | application name
+device_type | `STRING` | device type
+
+* cURL Example:
+
+```
+curl -X POST 'https://api.resin.io/v1/application' 
+-H 'Content-Type: application/json' 
+-H 'Authorization: Bearer <auth_token>' 
+--data '{"app_name":"<name>", 
+		  "device_type":"<device type>"}'
+```
+
+*The available device types are:*
```

Comment:
It would be nice to link this to the resin-os org, or for their individual repos maybe